internal/db/db.go

```diff
@@ -39,24 +40,73 @@ func (db *DB) Open(bootstrap bool, project string) error {
 	if !bootstrap {
 		checkVersions := func(ctx context.Context, current int, tx *sql.Tx) error {
 			schemaVersion := newSchema.Version()
-			err = cluster.UpdateClusterMemberSchemaVersion(tx, schemaVersion, db.listenAddr.URL.Host)
+			internalExtensionsDB, externalExtensionsDB := ext.SerializeForDB()
+			info := cluster.InternalClusterMemberVersioningInfo{
+				SchemaVersion:         schemaVersion,
+				InternalAPIExtensions: internalExtensionsDB,
+				ExternalAPIExtensions: externalExtensionsDB,
+			}
+
+			err = cluster.UpdateClusterMemberSchemaVersionAndAPIExtensions(tx, info, db.listenAddr.URL.Host)
 			if err != nil {
 				return fmt.Errorf("Failed to update schema version when joining cluster: %w", err)
 			}
 
-			versions, err := cluster.GetClusterMemberSchemaVersions(ctx, tx)
+			versioningInfo, err := cluster.GetClusterMemberSchemaVersionsAndAPIExtensions(ctx, tx)
 			if err != nil {
 				return fmt.Errorf("Failed to get other members' schema versions: %w", err)
 			}
 
-			for _, version := range versions {
-				if schemaVersion == version {
-					// Versions are equal, there's hope for the
+			for _, versionInfo := range versioningInfo {
+				lInternal, remLInternal := extensions.IsSubset(internalExtensionsDB, versionInfo.InternalAPIExtensions) // this node's internal extensions are a subset of the other node's internal extensions
```

Comment:
I like this approach! But in the case of two nodes having the same number of extensions but not exactly the same ones, comparing the length of the extensions would not work right (maybe that this scenario will never happen in practice..)? That's why I tried to compare extension set rather than the number of extensions.. It might be overengineered. But let's keep things simple: I'll do as you suggested ;)

Reply:
fixed